File: HW 9/Python/Main.py
from Solution import Solution
from TreeNode import TreeNode

if __name__ == "__main__":
    root = TreeNode(4)
    root.left = TreeNode(2)
    root.right = TreeNode(7)
    root.left.left = TreeNode(1)
    root.left.right = TreeNode(3)
    
    solution = Solution()
    p = TreeNode(3)
    q = TreeNode(1)
    lca = solution.lowestCommonAncestor(root, p, q)
    print("Lowest common ancestor:", lca.val)

File: HW 9/Java/Main.java
public class Main {
    public static void main(String[] args) {
        TreeNode root = new TreeNode(4);
        root.left = new TreeNode(2);
        root.right = new TreeNode(7);
        root.left.left = new TreeNode(1);
        root.left.right = new TreeNode(3);
        
        Solution solution = new Solution();
        TreeNode p = new TreeNode(3);
        TreeNode q = new TreeNode(1);
        TreeNode lca = solution.lowestCommonAncestor(root, p, q);
        System.out.println("Lowest common ancestor: " + lca.val);
    }
}
